pkg/wallet/wallet.go

```diff
@@ -192,11 +202,23 @@ func Imported(nickname string, privateKey string) (*Wallet, error) {
 		return nil, fmt.Errorf("encoding private key B58: %w", err)
 	}
 
+	wallets, err := LoadAll()
+	ListOfAddresses := GetListOfAddresses(wallets)
+
+	if err != nil {
+		return nil, fmt.Errorf("loading wallets error: %w", err)
+	}
+
 	keypair := ed25519.NewKeyFromSeed(privKeyBytes)
 
 	pubKeyBytes := reflect.ValueOf(keypair.Public()).Bytes() // force conversion to byte array
 
 	addr := blake3.Sum256(pubKeyBytes)
+	Address := "A" + base58.CheckEncode(append(make([]byte, 1), addr[:]...))
+
+	if slices.Index(ListOfAddresses, Address) != -1 {
```

Comment:
I propose to use to remove GetListOfAddresses function and to use:

```slices.IndexFunc(wallets, func(wallet Wallet) bool { return wallet.Address == Address })```

it will iterate on the Wallet slice and test if any match with the provided Address